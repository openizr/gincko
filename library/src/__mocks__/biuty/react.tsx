/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import * as React from 'react';

/**
 * biuty/react mock.
 */
function Component(props: JSXElement): JSX.Element {
  const { onChange, onFocus, onClick } = props;
  const { transform, onKeyDown, onBlur } = props;

  // Covers `onChange` handler.
  setTimeout(() => {
    onChange?.('test');
    if (process.env.IS_DATE === 'true') {
      onChange?.('2020/02/20');
    }
  }, 10);

  // Covers `onFocus` handler.
  onFocus?.();

  // Covers `onBlur` handler.
  onBlur?.();

  // Covers `transform` function.
  transform?.('ok');
  transform?.('1002');
  transform?.('100220');
  transform?.('10022020');

  // Covers `onKeyDown` handler.
  onKeyDown?.('', { key: '1' });
  onKeyDown?.('', { key: 'A', ctrlKey: false, preventDefault: vi.fn() });
  // Covers `onClick` handler.
  if (process.env.NOCLICK !== 'true') {
    setTimeout(() => {
      onClick?.();
    }, 10);
  }
  return (
    <div>{JSON.stringify(props)}</div>
  );
}

export const UILink = Component;
export const UIButton = Component;
export const UIOptions = Component;
export const UITextfield = Component;
export const UITextarea = Component;
export const UIFilePicker = Component;
export const buildClass = vi.fn((...values: string[]): string => values.join(' '));
export const markdown = vi.fn((label: string, lightMode: boolean) => `MARKDOWN FOR ${label}, ${lightMode}`);
