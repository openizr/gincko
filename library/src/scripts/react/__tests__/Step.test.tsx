/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @vitest-environment jsdom
 */

import React from 'react';
import Step from 'scripts/react/Step';
import { render } from '@testing-library/react';

vi.mock('biuty/react');
vi.mock('scripts/core/Engine');
vi.mock('scripts/react/Field');

describe('react/Step', () => {
  const customComponents = {};
  const userInputs = { test: 'ok' };
  const variables = { var: 'value' };
  const i18n = vi.fn(() => 'TRANSLATED LABEL');
  const onUserAction = vi.fn() as unknown as OnUserAction;

  beforeEach(() => {
    vi.clearAllMocks();
  });

  test('renders correctly - active step', async () => {
    const { container } = render(<Step
      index={0}
      isActive
      step={{
        id: 'step',
        status: 'success',
        fields: [{
          id: 'Message',
          status: 'initial',
          component: 'Message',
          componentProps: {},
        }],
      }}
      i18n={i18n}
      variables={variables}
      userInputs={userInputs}
      onUserAction={onUserAction}
      customComponents={customComponents}
    />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - inactive step', async () => {
    const { container } = render(<Step
      index={0}
      step={{
        id: 'step',
        status: 'success',
        fields: [{
          id: 'Message',
          status: 'initial',
          component: 'Message',
          componentProps: {},
        }, null],
      }}
      i18n={i18n}
      isActive={false}
      variables={variables}
      userInputs={userInputs}
      onUserAction={onUserAction}
      customComponents={customComponents}
    />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
