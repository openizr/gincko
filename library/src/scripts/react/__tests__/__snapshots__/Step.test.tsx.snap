// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`react/Step > renders correctly - active step 1`] = `
<div
  class="gincko__step success step active"
  id="step__0"
>
  <div
    class="gincko__step__fields"
  >
    <div
      id="field"
    >
      {"field":{"id":"Message","status":"initial","component":"Message","componentProps":{}},"isActive":true,"variables":{"var":"value"},"userInputs":{"test":"ok"},"customComponents":{},"path":"step.0.Message"}
    </div>
  </div>
</div>
`;

exports[`react/Step > renders correctly - inactive step 1`] = `
<div
  class="gincko__step success step "
  id="step__0"
>
  <div
    class="gincko__step__fields"
  >
    <div
      id="field"
    >
      {"field":{"id":"Message","status":"initial","component":"Message","componentProps":{}},"isActive":false,"variables":{"var":"value"},"userInputs":{"test":"ok"},"customComponents":{},"path":"step.0.Message"}
    </div>
  </div>
</div>
`;
